desc/objectives/_free_boundary.py

```diff
@@ -0,0 +1,945 @@
+from desc.compute import get_params, get_profiles, get_transforms
+from desc.grid import LinearGrid
+from desc.nestor import Nestor
+from desc.objectives.objective_funs import _Objective
+from desc.singularities import (
+    DFTInterpolator,
+    FFTInterpolator,
+    virtual_casing_biot_savart,
+)
+from desc.utils import Timer, errorif, warnif
+
+from .normalization import compute_scaling_factors
+
+
+class VacuumBoundaryError(_Objective):
+    """Target for free boundary conditions on LCFS for vacuum equilibrium.
+
+    Computes the residuals of the following:
+
+    𝐁ₒᵤₜ ⋅ 𝐧 = 0
+    𝐁ₒᵤₜ² - 𝐁ᵢₙ² = 0
+
+    Where 𝐁ᵢₙ is the total field inside the LCFS (from fixed boundary calculation)
+    𝐁ₒᵤₜ is the total field outside the LCFS (from coils)
+    𝐧 is the outward surface normal.
+
+    (Technically for vacuum equilibria the second condition is redundant with the first,
+    but including it makes things more robust).
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium that will be optimized to satisfy the Objective.
+    ext_field : MagneticField
+        External field produced by coils.
+    target : float, ndarray, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        len(target) must be equal to Objective.dim_f
+    bounds : tuple, optional
+        Lower and upper bounds on the objective. Overrides target.
+        len(bounds[0]) and len(bounds[1]) must be equal to Objective.dim_f
+    weight : float, ndarray, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        len(weight) must be equal to Objective.dim_f
+    normalize : bool
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate error at. Should be at rho=1.
+    field_grid : Grid, optional
+        Grid used to discretize ext_field.
+    name : str
+        Name of the objective function.
+
+    """
+
+    _scalar = False
+    _linear = False
+    _print_value_fmt = "Boundary Error: {:10.3e} "
+    _units = "(mixed units)"
```

Comment:
List the different units, like `("(T)", "(T^2)")` or whatever they are. I don't think this gets used since you override it in the custom `print` method anyways, but it's nice as documentation of what the units are.